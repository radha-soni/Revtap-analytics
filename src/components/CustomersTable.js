import React from "react";
import { withStyles, makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import Paginate from "./Paginate";
import "./chart.css";

const StyledTableCell = withStyles((theme) => ({
  head: {
    color: "#676262",
  },
  body: {
    fontSize: 14,
  },
}))(TableCell);

const StyledTableRow = withStyles((theme) => ({
  root: {
    "&:nth-of-type(odd)": {},
  },
}))(TableRow);

const useStyles = makeStyles({
  tableContainer: { borderRadius: 0, width: "1280px", margin: "30px auto" },
  table: {
    margin: "20px auto",
    maxWidth: "1200px",
    borderRadius: 0,
  },

  header: { fontSize: 20, borderRadius: 0 },
  row: { backgroundColor: "#e0e1e8" },
  para: {
    paddingLeft: 40,
  },
});

export default function CustomersTable({
  currentCustomers,
  customerPerPage,
  totalCustomers,
  setCurrentPage,
}) {
  const classes = useStyles();

  return (
    <Paper className={classes.tableContainer}>
      <Paper variant="outlined" className={classes.header}>
        <p className={classes.para}>Customers</p>
      </Paper>

      <TableContainer className={classes.table} component={Paper} elevation={0}>
        <Table aria-label="customized table">
          <TableHead>
            <TableRow className={classes.row}>
              <StyledTableCell>ID</StyledTableCell>
              <StyledTableCell>First Name</StyledTableCell>
              <StyledTableCell>Last Name</StyledTableCell>
              <StyledTableCell>Email</StyledTableCell>
              <StyledTableCell>Created</StyledTableCell>
              <StyledTableCell>Orders</StyledTableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {currentCustomers &&
              currentCustomers.map((row) => (
                <StyledTableRow key={row.id}>
                  <StyledTableCell component="th" scope="row">
                    {row.id}
                  </StyledTableCell>
                  <StyledTableCell>{row.firstName}</StyledTableCell>
                  <StyledTableCell>{row.lastName}</StyledTableCell>
                  <StyledTableCell>{row.email}</StyledTableCell>
                  <StyledTableCell>{row.created}</StyledTableCell>
                  <StyledTableCell>{row.orders}</StyledTableCell>
                </StyledTableRow>
              ))}
          </TableBody>
        </Table>
      </TableContainer>
      <div className="paginate">
        <Paginate
          customerPerPage={customerPerPage}
          totalCustomers={totalCustomers}
          setCurrentPage={setCurrentPage}
        />
      </div>
    </Paper>
  );
}
