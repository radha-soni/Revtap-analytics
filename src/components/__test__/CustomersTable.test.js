import React from "react";
import renderer from "react-test-renderer";
import CustomersTable from "../CustomersTable";
test("Customer data renders correctly", () => {
  const component = renderer.create(<CustomersTable />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
