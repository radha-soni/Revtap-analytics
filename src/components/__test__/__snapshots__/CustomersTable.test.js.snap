// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer data renders correctly 1`] = `
<div
  className="MuiPaper-root makeStyles-tableContainer-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiPaper-root makeStyles-header-3 MuiPaper-outlined MuiPaper-rounded"
  >
    <p
      className="makeStyles-para-5"
    >
      Customers
    </p>
  </div>
  <div
    className="MuiPaper-root MuiTableContainer-root makeStyles-table-2 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <table
      aria-label="customized table"
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root makeStyles-row-4 MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-43"
            scope="col"
          >
            ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-43"
            scope="col"
          >
            First Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-43"
            scope="col"
          >
            Last Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-43"
            scope="col"
          >
            Email
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-43"
            scope="col"
          >
            Created
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-43"
            scope="col"
          >
            Orders
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
  <div
    className="paginate"
  >
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root"
    >
      <ul
        className="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
